src/main/java/org/mockito/internal/progress/MockingProgress.java

```diff
@@ -39,9 +42,30 @@
     
     void mockingStarted(Object mock, MockCreationSettings settings);
 
-    void addListener(MockitoListener listener);
+    /**
+     * Registers the given Listener that will be notified when a mock object was created, see {@link #mockingStarted(Object, MockCreationSettings)}. 
+     * The listener must provide a void method annotated with {@link Subscribe} and a sole 
+     * {@link MockCreatedReport} parameter or implement the interface {@link MockCreationListener} . 
+     * Here is an example:
+     *  <pre>
+     *  Object listener =  new Object(){
+     *      public void logCreatedMocks(MockCreatedReport report){
+     *         logger.log(report);
+     *      }
+     *  };
+     *  </pre>
+     *  <p>
+     *  Note: Listeners implementing the interface {@link MockCreationListener} will be notified via {@link MockCreationListener#onMockCreated(Object, MockCreationSettings)}.  
+     * 
+     * @param listener must not be <code>null</code>
+     */
+    void addListener(Object listener);
 
-    void removeListener(MockitoListener listener);
+    /**
+     * Unregisters the given listener. It will not be notified about mock creations anymore.
```

Comment:
Yes, currently only MockCreationListener extends MockitoLister, other listener interfaces can be added in the same way.